database/migrations/community/2024_12_16_000002_update_forumtopic_table.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\SQLiteConnection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table('ForumTopic', function (Blueprint $table) {
+            if (DB::connection() instanceof SQLiteConnection) {
+                $table->dropForeign(['ForumID']);
+            } else {
+                $table->dropForeign('forum_topics_forumid_foreign');
+            }
+        });
+        Schema::table('ForumTopic', function (Blueprint $table) {
+            if (DB::connection() instanceof SQLiteConnection) {
+                // do nothing
+            } else {
+                $table->dropIndex('forum_topics_forum_id_index');
+            }
+        });
+        Schema::table('ForumTopic', function (Blueprint $table) {
+            if (DB::connection() instanceof SQLiteConnection) {
+                $table->dropForeign(['author_id']);
+            } else {
+                $table->dropForeign('forumtopic_author_id_foreign');
+            }
+        });
+
+        Schema::rename('ForumTopic', 'forum_topics');
+
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->renameColumn('ID', 'id');
+            $table->renameColumn('ForumID', 'forum_id');
+            $table->renameColumn('Title', 'title');
+            $table->renameColumn('DateCreated', 'created_at');
+            $table->renameColumn('Updated', 'updated_at');
+            $table->renameColumn('LatestCommentID', 'latest_comment_id');
+            $table->renameColumn('RequiredPermissions', 'required_permissions');
+        });
+
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->text('body')->after('title')->nullable();
+        });
+
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->index('forum_id', 'idx_forum_topics_forum_id'); // use a custom name to make SQLite happy
+        });
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->foreign('forum_id')->references('id')->on('forums')->onDelete('set null');
+        });
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->foreign('author_id')->references('ID')->on('UserAccounts')->onDelete('set null');
+        });
+    }
+
+    public function down(): void
+    {
+        Schema::table('forum_topics', function (Blueprint $table) {
+            $table->dropForeign(['forum_id']);
+            $table->dropForeign(['author_id']);
+            $table->dropIndex(['forum_id']);
```

Comment:
This is failing. Perhaps because of the custom name?
```
  SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `forum_topics_forum_id_index`; check that it exists (Connection: mysql, SQL: alter table `forum_topics` drop index `forum_topics_forum_id_index`)
```

Reply:
Lots of issues on my local from migrating / rolling back frequently while testing these changes. I think this should now be fixed in latest.